### DISCUSSION TITLE
Trying to add new row as an 'admin user': "new row violates row-level security policy for table "recipes""

### DISCUSSION THREAD

**Discussion by phillipmohr** [supabase/supabase#9943] [Questions] [answered]
- "nuxt": "^3.0.0-rc.12",
- "@nuxtjs/supabase": "^0.1.25",

hey, I created an admin backend where you can only get access when `claims_admin` is `true` inside the user's `app_metadata`. I followed [this tutorial](https://github.com/supabase-community/supabase-custom-claims) to accomplish that.

Now I'm trying to only allow admin users to add a new recipe (to my table `recipes`). 
I followed [this part](https://github.com/supabase-community/supabase-custom-claims#only-allow-users-with-claim_admin--true) of the tutorial to accomplish that, but unfortunately it's not working. 

My policy looks like this
<img width="1113" alt="Bildschirmfoto 2022-10-28 um 17 30 33" src="https://user-images.githubusercontent.com/30236221/198675800-3e6c2fd5-487b-4a63-963d-a57e590d024e.png">

console.log of user shows me that `claims_admin` is `true`, but I'm getting `"new row violates row-level security policy for table "recipes"" `
![Uploading Bildschirmfoto 2022-10-28 um 17.31.55.png…]()

This is how I try to insert a new row
```js
const { error } = await supabase
        .from('recipes')
        .insert(data)
```

What am I doing wrong? 

Thanks for any help!

**Comment by GaryAustin1** [comment, marked as answer]
Not sure on the nuxtjs client details, but that looks like v1 code.  With v1 clients normally insert also does a select unless you use returning minimal parameter.  
Does the user meet the select policy for that table?

**Reply by phillipmohr** [reply]
Yes, that was it! Thank you very much :)
